Location field made more visually appealing, enahnced ui-ux

ranisa-gupta16 opened the pull request:
## JIRA Ticket


## Description

- Made location field to incorporate fieldwrapper for consistent styling and formatting with rest of the form fields
-Improved button styling for better visibility & responsiveness
-Added, Larger map icon with proper spacing from the text
-Added hover effects & transition animations for better UX
- Animated checkmark appears when a location is successfully selected
- Dynamic color changes when a selection is missing (red cross feedback)
- Adjusted spacing & text formatting for better clarity
- Responsiveness added

Updated fucntionality:
- If user click save, lat long will be saved and displayed
- If map selection is cleared, the lat long will be updated with alert that its been cleared
- If map point was selected , user clickmap again but close it, nothing gets updated for lat/long, user will be given the message that map point selection was cancelled.

Refer to screenshots below to see how its working!


## How to Test
PLease view Location button in survey 

Screenshots:
- No map point has been selected
![image](https://github.com/user-attachments/assets/c16bcb40-d674-4cd5-a94f-9c610cc60900)

- Map has additional action  - To be able to clear the selection 
![image](https://github.com/user-attachments/assets/0a401454-65d6-4914-b595-e7adc42df834)

- Map point is selected, lat/long is updated with green check
![image](https://github.com/user-attachments/assets/b19dac8c-9fff-4d6f-973a-b73011d93be1)

- Map point was selected again but user clicked close, the lat/long wont' be udpated
![image](https://github.com/user-attachments/assets/e0ece815-2844-4bcd-a4d7-211cb863699c)

- Map point was selected but user click clear button to clear the point
![image](https://github.com/user-attachments/assets/2f0dbf78-8a79-4065-b003-4f7569697f43)



## Checklist

- [x] I have confirmed all commits have been signed.
- [x] I have added JSDoc style comments to any new functions or classes.
- [x] Relevant documentation such as READMEs, guides, and class comments are updated.

ranisa-gupta16 commented:
Note - Action for ranisa  - to improve map point cancel and save functionality.
Add a clear action to the mpa too for the user to clear the selected point.

ranisa-gupta16 commented:
> Note - Action for ranisa - to improve map point cancel and save functionality. Add a clear action to the mpa too for the user to clear the selected point.

This functionality has been now updated!

stevecassidy commented on app/src/gui/fields/maps/MapWrapper.tsx:
```diff
@@ -260,29 +278,104 @@ function MapWrapper(props: MapProps) {
-            <Button autoFocus color="inherit" onClick={handleClose}>
-              save
+
+            <Button
```
These buttons are both green in the Fieldmark theme. (Clear and Save)

ranisa-gupta16 replied:
Noted!

stevecassidy reviewed (commented):
Good apart from some theme problems and a bit of wording.

ranisa-gupta16 commented on app/src/gui/fields/maps/MapWrapper.tsx:
```diff
@@ -225,12 +234,21 @@ function MapWrapper(props: MapProps) {
+            // Only update lat/long if "save" is clicked
+            props.callbackFn(geoJsonFeatures, 'save');
+            notify.showSuccess('Location saved successfully!');
+          } else if (action === 'clear') {
```
No worries, I'll update this functionality to prompt user to choose after they clear it.

ranisa-gupta16 commented on app/src/gui/fields/maps/MapFormField.tsx:
```diff
@@ -145,37 +209,85 @@ export function MapFormField({
+          >
+            {isLocationSelected
+              ? valueText
+              : 'No location selected, click above to choose a point!'}
```
Noted!

ranisa-gupta16 commented on app/src/gui/fields/maps/MapWrapper.tsx:
```diff
@@ -260,29 +278,104 @@ function MapWrapper(props: MapProps) {
     <div>
-      <Button variant="outlined" fullWidth={true} onClick={handleClickOpen}>
-        {props.label}
+      <Button
```
Yup that's right, theme issue, I'll test with default theme and use appropriate theme colors for this to prevent inconsistency in theme. :-)

ranisa-gupta16 commented:
Updated  functionality as per above comments, and theme  for both- bss and default:
@stevecassidy 

Screenshots for reference: 

BSS THEME:
![image](https://github.com/user-attachments/assets/40e167da-86a1-44de-84a7-b00fe60e3a0e)
![image](https://github.com/user-attachments/assets/7ea2c467-5894-4cae-9fce-66006e77fb65)
![image](https://github.com/user-attachments/assets/07de648b-3d9a-4684-943e-ee403d53aa35)
![image](https://github.com/user-attachments/assets/53d954d2-dc0a-4cc8-9d8d-5d697f035a3e)
![image](https://github.com/user-attachments/assets/095a13dd-ec3e-4a80-a7a7-2d8d03e95c2d)



DEFAULT THEME:
![image](https://github.com/user-attachments/assets/7fa6257b-99ef-4b3c-ab1e-bc0619767f1d)
![image](https://github.com/user-attachments/assets/52f78b08-39aa-488a-88bb-eeec24f07fa6)
![image](https://github.com/user-attachments/assets/19c1882c-8601-4290-bffa-6d10b23de71b)
![image](https://github.com/user-attachments/assets/5899aa37-b977-4606-992c-a0151f861d64)
![image](https://github.com/user-attachments/assets/14121c16-7d1e-4415-8e93-7d7756c6cbed)
![image](https://github.com/user-attachments/assets/7bf45cc0-e80d-408d-99dc-adfd70bb3a5d)

PeterBaker0 reviewed (changes requested):
Overall it is more visually striking, but I think the look/feel seems a bit out of alignment with the new form styling approach (i.e. minimalist, clear structure, not a lot of colours). 

however, it is a big improvement so I am happy to go with this styling approach for now. 

There are a couple of issues with the functionality though 

a) you can submit no location selection which results in this state 
![image](https://github.com/user-attachments/assets/cb5602d8-5a32-4f3a-9e67-8ff1f6c50b2a)
 
 I think a user would find that confusing - you shouldn't be able to save no location selection.  You can also get into the above state by clearing the selection.
 
 b) I think there are excessive pop-ups. No where else in the app do we have popups to confirm form functionality. Maybe just the clear yes/no dialog is sufficient - in my opinion there is no need to have user alerts/popups every time something works as anticipated.

ranisa-gupta16 commented:
This PR is ready for re-review
Updated screenshots for reference:
- 
![image](https://github.com/user-attachments/assets/4383d0cb-3585-49d2-a0d5-c4c221dd8a0d)
![image](https://github.com/user-attachments/assets/28fb58a3-67a9-45d7-a88b-7a4518d1530b)
![image](https://github.com/user-attachments/assets/ebca15f7-a941-463f-b815-e8f85b7412cb)
![image](https://github.com/user-attachments/assets/505e97a6-167e-423f-87d8-b232ef034d26)
![image](https://github.com/user-attachments/assets/f5bb1943-514c-4d74-a592-e7b93c155d37)

For the  placement of the edit icon for the map interaction. Initially, we were trying to manage the edit functionality across both MapWrapper and MapFormField, but we ran into complications due to the way these components are structured. The green checkmark indicating location selection is part of MapFormField, while all the state and logic for handling the map (including opening, clearing, and saving) reside inside MapWrapper. This separation made it difficult to properly sync actions between the two components without unnecessary back-and-forth prop drilling or complex state management. Let me know if any issues Thanks

ranisa-gupta16 commented:
> Overall it is more visually striking, but I think the look/feel seems a bit out of alignment with the new form styling approach (i.e. minimalist, clear structure, not a lot of colours).
> 
> however, it is a big improvement so I am happy to go with this styling approach for now.
> 
> There are a couple of issues with the functionality though
> 
> a) you can submit no location selection which results in this state ![image](https://private-user-images.githubusercontent.com/87056634/411366248-cb5602d8-5a32-4f3a-9e67-8ff1f6c50b2a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyNDQ4MzAsIm5iZiI6MTczOTI0NDUzMCwicGF0aCI6Ii84NzA1NjYzNC80MTEzNjYyNDgtY2I1NjAyZDgtNWEzMi00ZjNhLTllNjctOGZmMWY2YzUwYjJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjExVDAzMjg1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1MDUzY2I3NWRmZjdjMGRlN2M1MzA2ZjE5NWQ2YTAyYzVkY2MzMTMzYmMyMmY4YWM4OGRhMzQxNzQ5YzBhODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4oN-RCytwGzE-CNPBasZq8EAhbjtkZznePhEywsggQY)
> 
> I think a user would find that confusing - you shouldn't be able to save no location selection. You can also get into the above state by clearing the selection.
> 
> b) I think there are excessive pop-ups. No where else in the app do we have popups to confirm form functionality. Maybe just the clear yes/no dialog is sufficient - in my opinion there is no need to have user alerts/popups every time something works as anticipated.

-Addressed and updated changes as per discussion on monday.

PeterBaker0 reviewed (approved):
Much improved - have fixed the clear logic as discussed and renamed some stuff but good to go. Checked with both themes. I think we can unblock this given its a notable improvement from previous implementation - though possibly could be improved to be more stylistically consistent with existing look/feel through sizing/layout changes.

PeterBaker0 reviewed (approved):
Made some changes to repeated elements, unnecessary state variables and use effects, fixed the clear logic (which was still saving on close). 

Lgtm
